src/game/savegame/savegame_bson.c

```diff
@@ -797,6 +800,34 @@ static bool Savegame_BSON_LoadLara(
     return true;
 }
 
+static bool SaveGame_BSON_LoadCurrentMusic(struct json_object_s *music_obj)
+{
+    if (!g_Config.load_current_music) {
+        return true;
+    }
+
+    if (!music_obj) {
+        LOG_WARNING("Malformed save: invalid or missing current music");
+        return true;
+    }
+
+    int16_t current_track = json_object_get_int(music_obj, "current_track", -1);
+    int32_t timestamp_arr[2];
+    timestamp_arr[0] = json_object_get_int(music_obj, "timestamp1", 0);
+    timestamp_arr[1] = json_object_get_int(music_obj, "timestamp2", 0);
+    int64_t *timestamp = (int64_t *)timestamp_arr;
+    if (current_track) {
+        Music_Play(current_track);
+        if (!Music_SeekTimestamp(current_track, *timestamp)) {
```

Comment:
I've noticed that on load, you hear a very brief moment of the beginning of the track before it jumps to the restored timestamp. It's negligible imo, but I wonder if there is a way to seek before the music is started?

Reply:
Yes I noticed that too, but it's hard to tell since it's like a on frame thing it seems like. I'll look into it.

Reply:
I think that this might be caused by the flow:

- We call `Game_Start`
- Inside `Game_Start`, we call `Level_Initialise`
- `Level_Initialise` calls `Music_Play` which starts the audio track, probably in an asynchronous way
- Next up, inside `Game_Start` we call `Savegame_Load`
- This calls `Music_SeekTimestamp` as well as `Music_Play`

Could you verify if removing the call to `Music_Play` from inside `Level_Initialise` fixes the clipping issue?
If so, I suggest testing whether a fix like this would work:

```diff
diff --git a/src/game/game/game.c b/src/game/game/game.c
index 8f95ea29..c016ea69 100644
--- a/src/game/game/game.c
+++ b/src/game/game/game.c
@@ -133,6 +133,12 @@ bool Game_Start(int32_t level_num, GAMEFLOW_LEVEL_TYPE level_type)
         // reset current info to the defaults so that we do not do
         // Item_GlobalReplace in the inventory initialization routines too early
         Savegame_InitCurrentInfo();
+
+        // prevent audio clipping that occurs when level plays a music track,
+        // and later savegame routines override it while the stream is already
+        // playing in the background thread
+        Music_Pause();
+
         if (!Level_Initialise(level_num)) {
             return false;
         }
@@ -140,6 +146,8 @@ bool Game_Start(int32_t level_num, GAMEFLOW_LEVEL_TYPE level_type)
             LOG_ERROR("Failed to load save file!");
             return false;
         }
+        Music_Unpause();
+
         break;
 
     case GFL_RESTART:
```

Reply:
Ah I was confused at first but you mean `Level_Initialise` calls `Music_PlayLooped`.

I made this change and I think it does actually fix the issue! Will definitely need some others to test as well to confirm.

Reply:
It's definitely better, I think I do still hear it extremely briefly though - or am I imagining it?

https://youtu.be/kEjJxek18Cs

Reply:
I think I hear something but I'd say it's negligible and feels like a part of the melody, too.

Reply:
Yes it's hard to tell. I think it sounds pretty good now also.

Reply:
It sounds as though a snippet of the passport in/out SFX is repeated very briefly upon load. Is that possible?
It's super nice this issue is being worked on (!), but I do find the added noise unfortunate.

Reply:
It's hard to tell if it's just mid instrumental or not. It sounds OK on these to me? I'm not totally sure lol.

https://streamable.com/lj45x7

Reply:
Hmmm. Is it possible it's different now compared to the Folly clip yesterday? It sounds changed (improved!).

I've compiled all of McCree's TR1-3 music in my cloud some years ago and have since accumulated (too) many hours listening to it. Want to say I know all of its nuances decently well. Something sounds off in your above video, but it is pretty hard to tell. Can you post it on the Discord to get more opinions?

Is there a loudness spike for a split second until it adjusts, right after load?

Other times it reminds me of a cassette player resuming playback, and needing a very brief moment to reach full pitch.

In any case, I generally love how this is behaving now. Such a nice improvement 👏